File: src/backend/app/schemas/career_path.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class DepartmentBrief(BaseModel):
    id: uuid.UUID
    name: str
    model_config = {"from_attributes": True}


class CompetencyBrief(BaseModel):
    id: uuid.UUID
    name: str
    model_config = {"from_attributes": True}


class CareerPathRequirementInput(BaseModel):
    competency_id: uuid.UUID
    required_level: int
    is_mandatory: bool = False


class CareerPathRequirementRead(BaseModel):
    id: uuid.UUID
    career_path_id: uuid.UUID
    competency_id: uuid.UUID
    competency: CompetencyBrief
    required_level: int
    is_mandatory: bool
    model_config = {"from_attributes": True}


class CareerPathCreate(BaseModel):
    from_department_id: uuid.UUID
    to_department_id: uuid.UUID


class CareerPathRead(BaseModel):
    id: uuid.UUID
    from_department_id: uuid.UUID
    to_department_id: uuid.UUID
    from_department: DepartmentBrief
    to_department: DepartmentBrief
    is_active: bool
    requirements: list[CareerPathRequirementRead] = []
    created_at: datetime
    model_config = {"from_attributes": True}


class ReadinessItem(BaseModel):
    competency_id: uuid.UUID
    competency_name: str
    required_level: int
    is_mandatory: bool
    current_score: float | None
    gap: float | None
    is_met: bool


class CareerReadinessRead(BaseModel):
    career_path_id: uuid.UUID
    user_id: uuid.UUID
    is_ready: bool
    readiness_pct: float
    mandatory_met: bool
    items: list[ReadinessItem]

File: src/frontend/src/types/career_path.ts
export interface CareerPathRequirementInput {
  competency_id: string
  required_level: number
  is_mandatory?: boolean
}

export interface CareerPathRequirementRead {
  id: string
  career_path_id: string
  competency_id: string
  competency: { id: string; name: string }
  required_level: number
  is_mandatory: boolean
}

export interface CareerPathCreate {
  from_department_id: string
  to_department_id: string
}

export interface CareerPathRead {
  id: string
  from_department_id: string
  to_department_id: string
  from_department: { id: string; name: string }
  to_department: { id: string; name: string }
  is_active: boolean
  requirements: CareerPathRequirementRead[]
  created_at: string
}

export interface ReadinessItem {
  competency_id: string
  competency_name: string
  required_level: number
  is_mandatory: boolean
  current_score: number | null
  gap: number | null
  is_met: boolean
}

export interface CareerReadinessRead {
  career_path_id: string
  user_id: string
  is_ready: boolean
  readiness_pct: number
  mandatory_met: boolean
  items: ReadinessItem[]
}
